src/3d/reign.c

```diff
@@ -138,29 +159,126 @@ bool RE_instance_load(struct RE_instance *instance, const char *name)
 {
 	if (!instance)
 		return false;
-	if (instance->model)
-		model_free(instance->model);
+	unload_instance(instance);
 	instance->model = model_load(instance->plugin->aar, name, instance->plugin->renderer);
+	if (instance->model && instance->model->nr_bones > 0)
+		instance->bone_transforms = xcalloc(instance->model->nr_bones, sizeof(mat4));
 	return !!instance->model;
 }
 
-void RE_debug_print(struct RE_plugin *p)
+bool RE_instance_load_motion(struct RE_instance *instance, const char *name)
+{
+	if (!instance || !instance->model)
+		return false;
+	instance->motion = motion_load(name, instance->model, instance->plugin->aar);
```

Comment:
There is a memory leak here. If instance->motion isn't NULL, it gets overwritten. The leak happens in battles.

Reply:
Good catch, fixed.